packages/core/index.ts

```diff
@@ -97,6 +103,33 @@ function toGraphs(api: string) {
   }
 }
 
+async function applyUpdates(api: string, validDirs: string[], dataset: DatasetCore) {
+  const engine = new QueryEngine()
+  const store = new Store([...dataset])
+  const results = $rdf.dataset()
+  for (const dir of validDirs) {
+    for await (const file of walk(dir)) {
+      if (!file.endsWith('.ru')) {
+        continue
+      }
+      const destination = new Store()
+      const relative = path.relative(dir, file)
+      log.debug('Applying updates from %s, dataset size %s', relative, store.size)
+      const query = fs.readFileSync(file, 'utf-8')
+      for (const command of query.split(';')) {
+        await engine.queryVoid(command, {
+          sources: [destination, store],
+          destination,
+          baseIRI: baseIRI(relative, api),
+        })
+      }
+      log.debug('Applied updates from %s, dataset size %s', relative, store.size)
+      results.addAll(destination)
+    }
+  }
+  return $rdf.dataset([...dataset, ...results])
+}
```

Comment:
I think I would just move this to a separate module

Reply:
Interestingly, we also have to move the `baseUri` function that I extracted. Or maybe extracting that function was not a good idea in the first place, because we may want to evolve it differently in the two cases